### DISCUSSION TITLE
Possible undocumented change of behavior for the `SpatieMediaLibraryFileUpload` component

### DISCUSSION THREAD

**Discussion by speniti** [filamentphp/filament#3551] [Help]
Hi!
I've noticed a possible undocumented change of behavior for the `SpatieMediaLibraryFileUpload` component. 

In my project I have a form that contains only `SpatieMediaLibraryFileUpload` components and when I call the `fill` method without any argument on the form nothing happens, while previously the form was correctly filled with the data loaded by relations.
I've checked the source code of the admin panel and seems that the issue is related to this commit "[fix: Refactor hydration process](https://github.com/filamentphp/filament/commit/ecf798ed9df0961261f7c8ee199444ed26f5d7ae)" and specifically this [file](https://github.com/filamentphp/filament/blame/2.x/packages/forms/src/Components/Concerns/HasState.php#L148).

In my case, this is preventing the call to the `loadStateFromRelationships` method and my form remains blank even if I've called the `fill` method.

I've swapped the call to the `fill` method with a direct call to the `loadStateFromRelationships` method and the form is now working, but I'm wondering if this is correct since I haven't found any reference to the `loadStateFromRelationships` method in the docs.

**Comment by danharrin** [comment]
Can you please send your form code? We use that plugin in the demo, and as far as I know the images are still being filled fine

**Reply by speniti** [reply]
Sure! I've created a gist in order to make it easy to read, here's the link https://gist.github.com/speniti/ce6a5c1f9ed7209ad25694ef93dc7a51

**Reply by danharrin** [reply]
Hello! Just tried it out. `$this->form->fill([])` works for me. Make sure you're passing an array to `fill()` with your form data, otherwise the form will fill with default values, which of course will not contain your media attachments.

**Reply by danharrin** [reply]
FWIW, you also need a place to store all your form data. You could create a new public property for every form field, or just do this:

```php
public $data = [];

protected function getFormStatePath(): string 
{
    return 'data';
} 
```

https://filamentphp.com/docs/2.x/forms/getting-started#scoping-form-data-to-an-array-property
